src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.cs

```diff
@@ -108,26 +108,22 @@ internal override void OnBufferAllocated(byte[] buffer)
         {
             Debug.Assert(_preallocatedOverlapped == null);
 
-            _preallocatedOverlapped = new PreAllocatedOverlapped(FileStreamCompletionSource.s_ioCallback, this, buffer);
+            _preallocatedOverlapped = new PreAllocatedOverlapped(FileStreamValueTaskSource.s_ioCallback, this, buffer);
         }
 
-        SafeFileHandle IFileStreamCompletionSourceStrategy.FileHandle => _fileHandle;
-
-        FileStreamCompletionSource? IFileStreamCompletionSourceStrategy.CurrentOverlappedOwner => _currentOverlappedOwner;
-
-        FileStreamCompletionSource? IFileStreamCompletionSourceStrategy.CompareExchangeCurrentOverlappedOwner(FileStreamCompletionSource? newSource, FileStreamCompletionSource? existingSource)
```

Comment:
I really like that you have removed `IFileStreamCompletionSourceStrategy` and all the ugliness related to it. 👍